api/scenario_api/src/scenario_api_node.cpp

```diff
@@ -19,226 +19,237 @@
+        false));  // false: do not consider z-axis of pointcloud
     int current_id;
     double dist1, dist2, dist3;
     bool over_line;
-    if (obj.getCurrentLaneID(current_id)) ROS_INFO_STREAM("getCurrentLaneID:" << current_id);
-    if (obj.getDistanceToTrafficLight(34806, dist1)) {
-      ROS_INFO_STREAM("dist to traffic light: 34806 is " << dist1);
+    if (obj->getCurrentLaneID(current_id))
+      RCLCPP_INFO_STREAM(obj->get_logger(), "getCurrentLaneID:" << current_id);
+    if (obj->getDistanceToTrafficLight(34806, dist1)) {
+      RCLCPP_INFO_STREAM(obj->get_logger(), "dist to traffic light: 34806 is " << dist1);
     }
-    if (obj.getDistanceToTrafficLine(34806, dist2)) {
-      ROS_INFO_STREAM("dist to traffic light line: 34806 is " << dist2);
-    } else if (obj.getDistancefromCenterLine(dist3))
-      ROS_INFO_STREAM("getDistancefromCenterLine:" << dist3);
-    if (obj.checkOverTrafficLine(34806, over_line)) {
-      ROS_INFO_STREAM("over stop line (34806)? :" << over_line);
+    if (obj->getDistanceToTrafficLine(34806, dist2)) {
+      RCLCPP_INFO_STREAM(obj->get_logger(), "dist to traffic light line: 34806 is " << dist2);
+    } else if (obj->getDistancefromCenterLine(dist3))
+      RCLCPP_INFO_STREAM(obj->get_logger(), "getDistancefromCenterLine:" << dist3);
+    if (obj->checkOverTrafficLine(34806, over_line)) {
+      RCLCPP_INFO_STREAM(obj->get_logger(), "over stop line (34806)? :" << over_line);
     }
-    ROS_INFO_STREAM("isInLane:" << obj.isInLane());  // base_link is in lane or not
-    ROS_INFO_STREAM("LeftBlinker:" << obj.getLeftBlinker());
-    ROS_INFO_STREAM("RightBlinker:" << obj.getRightBlinker());
-    ROS_INFO_STREAM("WillLaneChange" << obj.willLaneChange());  // temporary implementation
-    Pose2D p = obj.getCurrentPose();
-    ROS_INFO_STREAM("selfpose:" << p.x << "," << p.y << "," << p.yaw);
-    geometry_msgs::PoseStamped p2 = obj.getCurrentPoseRos();
-    ROS_INFO_STREAM("selfpose:" << p2.pose.position << "," << p2.pose.orientation);
+    RCLCPP_INFO_STREAM(
+      obj->get_logger(), "isInLane:" << obj->isInLane());  // base_link is in lane or not
+    RCLCPP_INFO_STREAM(obj->get_logger(), "LeftBlinker:" << obj->getLeftBlinker());
+    RCLCPP_INFO_STREAM(obj->get_logger(), "RightBlinker:" << obj->getRightBlinker());
+    RCLCPP_INFO_STREAM(
+      obj->get_logger(), "WillLaneChange" << obj->willLaneChange());  // temporary implementation
+    Pose2D p = obj->getCurrentPose();
+    RCLCPP_INFO_STREAM(obj->get_logger(), "selfpose:" << p.x << "," << p.y << "," << p.yaw);
+    geometry_msgs::msg::PoseStamped p2 = obj->getCurrentPoseRos();
+    // TODO(fred-apex-ai) how to print msg to stdout? https://answers.ros.org/question/365776/how-to-print-a-message-to-stdout-in-ros2/
+    //    RCLCPP_INFO_STREAM(obj->get_logger(), "selfpose:" << p2.pose.position << "," << p2.pose.orientation);
+    RCLCPP_INFO(obj->get_logger(), "printing of selfpose currently unavailable");
```

Comment:
From the context of immediate usability and equivalence, I think it would be valuable to have a `parse_pose(const Pose & p)` lambda (with a TODO) to parse poses into strings which is to be replaced once the feature you posted is available.

Reply:
I added `operator<<` so the original code needs only minimal modification @cvasfi please resolve this conversation if you are happy

Reply:
@fred-apex-ai That looks pretty good to me. Thanks!